ocs_ci/ocs/cluster.py

```diff
@@ -632,6 +744,222 @@ def time_taken_to_complete_rebalance(self, timeout=600):
                 time_taken = time.time() - start_time
                 return (time_taken / 60)
 
+    def osd_health_check(self, count):
+        """
+        OSD health check based on pod count
+        Args:
+            count (int): Expected number of Osd pods
+        Raises:
+            OsdCountException: if osd pod count doesn't match
+
+        """
+        timeout = 10 * len(self.pods)
+        logger.info(f"Expected OSDs count= {count}")
+        try:
+            assert self.POD.wait_for_resource(
+                condition='Running', selector=self.osd_selector,
+                resource_count=count, timeout=timeout, sleep=3,
+            )
+
+            actual_osds = len(pod.get_osd_pods())
+
+            assert count == actual_osds, f"Expected OSD {count},  Got {actual_osds}"
+        except exceptions.TimeoutExpiredError as e:
+            logger.error(e)
+            raise exceptions.OsdCountException(
+                f"Failed to achieve desired Osd count"
+                f" {count}"
+            )
+
+    def mgr_health_check(self, count):
```

Comment:
I think we can club the health_check into single functions, since thy differ in selectors?

Reply:
I fell if we club it will be hard to get where it failed

Reply:
I think we can do it but maybe as follow on PR, resolving for now.